sdk/mobile-center/src/test/java/com/microsoft/azure/mobile/MobileCenterTest.java

```diff
@@ -796,6 +796,27 @@ public void uncaughtExceptionHandler() {
         ShutdownHelper.shutdown(10);
     }
 
+
+    @Test
+    public void uncaughtExceptionHandlerTimeout() throws Exception {
+
+        /* It behaves the same on timeout. */
+        Semaphore semaphore = mock(Semaphore.class);
+        whenNew(Semaphore.class).withAnyArguments().thenReturn(semaphore);
+        when(semaphore.tryAcquire(anyLong(), any(TimeUnit.class))).thenReturn(false);
+        uncaughtExceptionHandler();
+    }
+
+    @Test
+    public void uncaughtExceptionHandlerInterrupted() throws Exception {
```

Comment:
I test that the behavior is the same regardless we could flush the queue or not, it calls the original test method which already has the same asserts.
Could add verify Log but I am really not a big fan of testing log traces. Will do but I don't like it as it can break easily.